subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationContainer.java

```diff
@@ -186,10 +134,7 @@ private void copyAllTo(DefaultConfiguration detachedConfiguration, Dependency[]
     }
 
     private DefaultConfiguration newConfiguration(String name, ConfigurationsProvider detachedConfigurationsProvider, RootComponentMetadataBuilder componentMetadataBuilder) {
-        return instantiator.newInstance(DefaultConfiguration.class, context, name, detachedConfigurationsProvider, resolver, listenerManager,
-            dependencyMetaDataProvider, resolutionStrategyFactory, fileCollectionFactory, buildOperationExecutor, instantiator, artifactNotationParser,
-            capabilityNotationParser, attributesFactory, componentMetadataBuilder, documentationRegistry, userCodeApplicationContext,
-            context, projectStateRegistry, workerThreadRegistry, domainObjectCollectionFactory, calculatedValueContainerFactory);
+        return defaultConfigurationFactory.create(name, detachedConfigurationsProvider, resolutionStrategyFactory, componentMetadataBuilder);
```

Comment:
Similar comment here as in `DefaultConfiguration` it is safe to depend on the values of the fields which now live in the factory holding their expected values?

Previously, a client of this class could keep a reference to any of these fields when constructing a new `DefaultConfigurationContainer` and modify them externally (if they are mutable, I didn't look very deeply) between the construction of `this` and the call to `newConfiguration`, as this class doesn't make defensive copies or anything.  This might break the expectations for how `newConfiguration` is meant to work.  If that was a problem, it would previously have been contained entirely within this class - a violation of an implicit class invariant - and now it would be distributed between this class and the factory.  Not even sure if this is a real problem, just something to consider.

Reply:
If such expectations were expected to occur, I would presume that a test would ensure that behaviour continued to be maintained.

That being said, I did my best to be careful to ensure that the types that were fixed were passed as constructor arguments to the `DefaultConfigurationFactory`, and the things that varied (were manipulated) were passed to `create`.